packages/core/src/StateNode.ts

```diff
@@ -761,7 +762,8 @@ class StateNode<
       value: this.resolve(stateFromConfig.value),
       configuration,
       done: isInFinalState(configuration, this),
-      tags: getTagsFromConfiguration(configuration)
+      tags: getTagsFromConfiguration(configuration),
+      machine: (this.machine as unknown) as AnyStateMachine
```

Comment:
nit: right now `this.machine === this` so we could replace this with:
```suggestion
      machine: (this as unknown) as AnyStateMachine
```

Reply:
@Andarist There is a chance that `this` can refer to a `StateNode` instead of the root state node (machine)

Reply:
well, ye, but I can't imagine somebody doing something like this:
```js
machine.states.a.resolveState(/*...*/)
```

Either way is fine though